File: app/src/main/java/com/prey/barcodereader/BarcodeGraphicTracker.java
package com.prey.barcodereader;

import android.app.Activity;

import com.prey.PreyLogger;
import com.prey.barcodereader.ui.camera.GraphicOverlay;
import com.google.android.gms.vision.Detector;
import com.google.android.gms.vision.Tracker;
import com.google.android.gms.vision.barcode.Barcode;

class BarcodeGraphicTracker extends Tracker<Barcode> {
    private GraphicOverlay<BarcodeGraphic> mOverlay;
    private BarcodeGraphic mGraphic;
    private BarcodeCaptureActivity activity;

    BarcodeGraphicTracker(GraphicOverlay<BarcodeGraphic> overlay, BarcodeGraphic graphic,BarcodeCaptureActivity activity) {
        mOverlay = overlay;
        mGraphic = graphic;
        this.activity=activity;
    }

    @Override
    public void onNewItem(int id, Barcode item) {
        mGraphic.setId(id);
    }

    @Override
    public void onUpdate(Detector.Detections<Barcode> detectionResults, Barcode item) {
        mOverlay.add(mGraphic);
        mGraphic.updateItem(item);
        activity.updateBarcode(item);
    }

    @Override
    public void onMissing(Detector.Detections<Barcode> detectionResults) {
        mOverlay.remove(mGraphic);
    }

    @Override
    public void onDone() {
        mOverlay.remove(mGraphic);
    }
}

File: app/src/main/java/com/prey/barcodereader/BarcodeGraphicTracker.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.barcodereader

import com.google.android.gms.vision.Detector.Detections
import com.google.android.gms.vision.Tracker
import com.google.android.gms.vision.barcode.Barcode
import com.prey.barcodereader.ui.camera.GraphicOverlay

internal class BarcodeGraphicTracker(
    private val mOverlay: GraphicOverlay<BarcodeGraphic>,
    private val mGraphic: BarcodeGraphic,
    private val activity: BarcodeCaptureActivity
) :
    Tracker<Barcode?>() {
    override fun onNewItem(id: Int, item: Barcode?) {
        mGraphic.setId(id)
    }

    override fun onUpdate(detectionResults: Detections<Barcode?>, item: Barcode?) {
        mOverlay.add(mGraphic)
       // mGraphic.updateItem(item)
        activity.updateBarcode(item)
    }

    override fun onMissing(detectionResults: Detections<Barcode?>) {
        mOverlay.remove(mGraphic)
    }

    override fun onDone() {
        mOverlay.remove(mGraphic)
    }
}
